### ISSUE TITLE
Failed to deploy V2 charts in the downstream clusters in rancher deployment done using docker install.

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#28587]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
- run rancher:`master-a1acca8567ec3f51735d3c657b17e282979d0d9a-head` single install 
- provision a downstream cluster, in this case, it is a custom cluster 
- deploy the monitoring app from the dashboard 

**Result:**
- a pod for helm operator shows in the dashboard
<img width="1865" alt="Screen Shot 2020-08-26 at 5 07 31 PM" src="https://user-images.githubusercontent.com/6218999/91368897-1bd5fd00-e7bf-11ea-81e5-0bd9486c6677.png">

- but it does not exist in the cluster 
```
>  k get pods -n cattle-monitoring-system
No resources found in cattle-monitoring-system namespace.
```
- also failed to delete the pod from the dashboard, the error indicates it does not exist
<img width="357" alt="Screen Shot 2020-08-26 at 5 07 12 PM" src="https://user-images.githubusercontent.com/6218999/91368949-3b6d2580-e7bf-11ea-98d5-492cb20d86ed.png">

**Comment by jiaqiluo** [comment]
The same behavior for installing rancher-gatekeeper and rancher-istio on the same setup 

<img width="1634" alt="Screen Shot 2020-08-26 at 5 34 22 PM" src="https://user-images.githubusercontent.com/6218999/91370142-686f0780-e7c2-11ea-98ca-b285839ef6a2.png">

**Comment by aaronRancher** [comment]
Tested with Rancher server `rancher\rancher:master-head` `version b5e1ce703`

On an HA setup, was able to deploy rancher-gatekeeper on a DO cluster.
![Screen Shot 2020-08-27 at 10 18 44 AM](https://user-images.githubusercontent.com/57409052/91474694-699a4600-e84f-11ea-934c-8e675a5a2bf9.png)

**Comment by maggieliu** [comment]
related to: https://github.com/rancher/rancher/issues/28502

**Comment by cbron** [comment]
primary issue is: #28279

**Comment by jiaqiluo** [comment]
the bug still exist on `master-5d5ef3f8fcbf878a7f001e6c81681daf531fd0cb-head` 

- install Rancher with `--priveleged` in docker command 
- provision a cluster
- go to the cluster -> dashboard -> apps -> install monitoring v2

Results:

<img width="1557" alt="Screen Shot 2020-09-03 at 4 59 35 PM" src="https://user-images.githubusercontent.com/6218999/92184770-e193dd80-ee06-11ea-9ee1-1e1fb942ec18.png">

**Comment by cbron** [comment]
At least one issue here is that the helm deployment pods are being created in the parent rancher cluster and not the downstream cluster. Potentially unrelated: when you exec into helm pod it can reach the kube api server but kubeconfig can't.

**Comment by cbron** [comment]
New PR: https://github.com/rancher/rancher/pull/28755

**Comment by jiaqiluo** [comment]
This is blocked by this dashboard issue https://github.com/rancher/dashboard/issues/1195

will come back when that issue is addressed

**Comment by jiaqiluo** [comment]
the bugfix is validated on `master-6132dbb3b458acd60282bc2505566349e008115d-head`

Monitoring v2 is installed successfully in the downstream cluster in Rancher single-install setup

<img width="1852" alt="Screen Shot 2020-09-08 at 4 12 06 PM" src="https://user-images.githubusercontent.com/6218999/92536582-e71a6a80-f1ee-11ea-8ad0-f222e09c39bc.png">

<img width="1865" alt="Screen Shot 2020-09-08 at 4 17 12 PM" src="https://user-images.githubusercontent.com/6218999/92536539-cd792300-f1ee-11ea-9e99-396479743c16.png">
<img width="1411" alt="Screen Shot 2020-09-08 at 4 17 24 PM" src="https://user-images.githubusercontent.com/6218999/92536544-ceaa5000-f1ee-11ea-8c50-f6d1014aa158.png">
